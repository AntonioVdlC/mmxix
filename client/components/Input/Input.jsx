import React from "react";
import PropTypes from "prop-types";

import styles from "./Input.css";

const Input = ({
  label,
  name,
  placeholder,
  type,
  value,
  handleInputChange
}) => (
  <label className={styles.label}>
    <span>{label}</span>
    <input
      className={styles.inpupt}
      type={type}
      name={name}
      value={value}
      placeholder={placeholder}
      onChange={({ target: { value } }) => handleInputChange(value)}
    />
  </label>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired
};

export default Input;
