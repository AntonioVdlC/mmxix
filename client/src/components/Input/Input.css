@import "../../styles/vars.css";

.label {
  display: flex;
  flex-direction: column;
  margin-bottom: var(--form-label-margin-bottom);
}
.label span {
  font-family: var(--font-family-secondary);
  font-size: var(--font-size-s);
}
.label .input {
  padding: var(--form-input-padding);
  border: var(--form-input-border);
  border-radius: var(--form-input-border-radius);
  font-family: var(--font-family-secondary);
  font-size: var(--font-size-xl);
  text-align: center;
}
