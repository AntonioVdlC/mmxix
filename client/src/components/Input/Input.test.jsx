import React from "react";
import renderer from "react-test-renderer";

import Input from "./Input";

describe("Input", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Input
          label={""}
          name={"arabic"}
          placeholder={"2019"}
          type={"number"}
          value={2019}
          handleInputChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
