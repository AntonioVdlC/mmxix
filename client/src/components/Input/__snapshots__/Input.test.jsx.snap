// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<label
  className="label"
>
  <span>
    
  </span>
  <input
    className="input"
    name="arabic"
    onChange={[Function]}
    placeholder="2019"
    type="number"
    value={2019}
  />
</label>
`;
